lib/Rules/PasswordHistory.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ *
+ * @author Semih Serhat Karakaya <karakayasemi@itu.edu.tr>
+ * @copyright Copyright (c) 2018, ownCloud GmbH
+ * @license GPL-2.0
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\PasswordPolicy\Rules;
+
+use OCA\PasswordPolicy\Db\OldPasswordMapper;
+use OCP\IL10N;
+use OCP\Security\IHasher;
+
+class PasswordHistory extends Base {
+
+	/** @var OldPasswordMapper */
+	protected $mapper;
+	/** @var IHasher */
+	protected $hasher;
+
+	/**
+	 * @param IL10N $l10n
+	 * @param OldPasswordMapper $mapper
+	 * @param IHasher
+	 */
+	public function __construct(IL10N $l10n, OldPasswordMapper $mapper, IHasher $hasher) {
+		parent::__construct($l10n);
+		$this->mapper = $mapper;
+		$this->hasher = $hasher;
+	}
+
+	/**
+	 * @param string $password
+	 * @param int $val
+	 * @param string $uid
+	 * @throws \Exception
+	 */
+	public function verify($password, $val, $uid) {

```

Comment:
if uid is null just return here - this check is only reasonable if the uid is given

Reply:
I checked the uid is not empty in the engine, but also 
added an additional check to here.